drivers/serial/uart_rtt.c

```diff
@@ -72,49 +86,140 @@ static void uart_rtt_poll_out(struct device *dev, unsigned char c)
 	SEGGER_RTT_Write(ch, &c, 1);
 }
 
+#ifdef CONFIG_UART_ASYNC_API
+
+static int uart_rtt_callback_set(struct device *dev, uart_callback_t callback,
+				 void *user_data)
+{
+	get_dev_data(dev)->callback = callback;
+	get_dev_data(dev)->user_data = user_data;
+	return 0;
+}
+
+static int uart_rtt_tx(struct device *dev, const uint8_t *buf, size_t len,
+		       int32_t timeout)
+{
+	const struct uart_rtt_config *cfg = get_dev_config(dev);
+	struct uart_rtt_data *data = get_dev_data(dev);
+	unsigned int ch = cfg ? cfg->channel : 0;
+
+	ARG_UNUSED(timeout);
```

Comment:
If timeout is not supported maybe tx should check for timeout being K_FOREVER and return an error in any other case?

Reply:
The docs for `uart_tx` say that the timeout parameter is only valid when flow control is enabled.
Given that flow control is never enabled, there doesn't seem to be a reason to do anything with a permanently invalid parameter.